sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleSummaryAggregator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoubleSummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class DoubleSummaryAggregator extends AbstractAggregator {
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      new AggregatorFactory() {
+        @Override
+        public Aggregator getAggregator() {
+          return new DoubleSummaryAggregator();
+        }
+      };
+
+  // The current value. This controls its own internal thread-safety via method access. Don't
+  // try to use its fields directly.
+  private final DoubleSummary current = new DoubleSummary();
+
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  @Override
+  void doMergeAndReset(Aggregator target) {
+    DoubleSummaryAggregator other = (DoubleSummaryAggregator) target;
+
+    DoubleSummary copy = current.copyAndReset();
+    other.current.update(copy.count, copy.sum, copy.min, copy.max);
+  }
+
+  @Nullable
+  @Override
+  public Point toPoint(long startEpochNanos, long epochNanos, Map<String, String> labels) {
+    return current.toPoint(startEpochNanos, epochNanos, labels);
+  }
+
+  @Override
+  public void recordDouble(double value) {
+    current.record(value);
+  }
+
+  private static class DoubleSummary {
```

Comment:
see above. Not sure what it buys us, and it adds a bit of clutter.

Reply:
This can enable the compiler/runtime to optimize this by knowing that it does not need to read the vtable to resolve the method pointer (at least in C++ can be done that way).

Reply:
That's *sort of* true. But, in this case, the aggregator is already built on an interface hierarchy that the JIT has to figure out already, so whether this is final or not is pretty moot. In java, there is no compile-time optimization (with very few exceptions, like string concatenation); it all happens in the JIT by doing analysis of the running system. Adding final here won't hurt anything, but it also won't help much, and it does add visual clutter to the code.